### ISSUE TITLE
md-autocomplete: infinite progress bar after clearing with cached queries

### ISSUE THREAD

**Issue by achaussende** [angular/material#8301]
**Actual behavior**:
-  The loading bar under the input is appearing and stay for infinite amount of time after clearing the md-autocomplete.
-  The loading bar should not appear.

**Steps to reproduce the issue**: 
You can reproduce it on the md-autocomplete basic usage demo:
- Select *Simulate query* for results
- Uncheck *Disable caching of queries*
- Select one item
- Clear

**CodePen**:
[CodePen](http://codepen.io/achaussende/pen/eZQGJy)

**Angular Versions**:
I spotted this issue on my own app:
-  Angular Version: 1.4.10
-  Angular Material Version: 1.0.8 but the issue is still on master version. (cf demo)

**Additional Information**:
-  Browser Type: Chrome
-  Browser Version: 50.0.2661.94 m
-  OS: Windows 8.1

NB: The issue does not happen in release 1.0.0 of Angular Material.
It does not happen with md-no-cache="true" either.

**Comment by Wsiegenthaler** [comment]
I also ran across this last week with 1.07 and Angular 1.5.3.  For now I've just hidden the progress indicator but this is only acceptable because my queries are always local.

`md-autocomplete md-progress-linear { display: none; }`

A couple things I've noticed which may or may not be relevant:
* The progress bar also appears when caching is disabled though it disappears again after a split second.
* This issue also occurs when the field is cleared by pressing escape.
